functions/api/action.js

```diff
@@ -203,6 +203,66 @@ router.post('/transferProduct', async (req, res, next) => {
     res.status(201).send({ transactionId: pendingRequestsId, toBranchTransactionId, fromBranchTransactionId })
 })
 
+/**
+ * Route to move products from headoffice to a branch(without request)
+ * @returns 201 Created
+ */
+router.post('/moveProduct', async (req, res, next) => {
+    const { error } = validateParams(req.body, constants.MOVE)
+    if (error) {
+        const err = new Error(error.details[0].message)
+        err.statusCode = 400
+        next(err)
+        return;
+    }
+    const { toBranch, fromBranch, toBranchName, fromBranchName, product, productName, operationalQuantity, note } = req.body
+
+    //check if branches exist
+    const toBranchRef = db.collection(constants.BRANCHES).doc(toBranch);
+    const fromBranchRef = db.collection(constants.BRANCHES).doc(fromBranch);
+    const docs = await db.getAll(toBranchRef, fromBranchRef)
+    const toBranchDoc = docs[0]
+    const fromBranchDoc = docs[1]
+
+    if (!toBranchDoc.exists || !fromBranchDoc.exists) {
+        const error = new Error(`Branch does not exist. Can not transfer out products`)
+        error.statusCode = 404
+        next(error)
+        return
+    }
+
+    //create transferOut transaction
+    const fromBranchData = {
+        date: new Date(),
+        branch: fromBranch,
+        operationalQuantity,
+        product, 
+        productName,
+        note,
+        toBranchName,
+        user: req.user.email
+    }
+    fromBranchTransaction = createTransaction(fromBranchData)
+
+    //create transferIn transaction
+    const toBranchData = {
+        date: new Date(),
+        branch: toBranch,
+        operationalQuantity,
+        product, 
+        productName,
+        note,
+        fromBranchName,
+        user: req.user.email
+    }
+    tobranchTransaction = createTransaction(toBranchData)
+
+    const fromBranchTransactionId = await fromBranchTransaction
```

Comment:
qq: why do we need separate statements for transactionID's?
Can we do like  const fromBranchTransactionId = createTransaction(fromBranchData)?

Reply:
so we want to create two trxs - transferIn and transferOut. Instead of doing Promise.all, this is another way. When I call createTransaction,two requests will fire for both transferIn and transferOut almost at the same time, later will await(wait for both to complete)